lib/config.go

```diff
@@ -425,3 +426,11 @@ func SignedTemplatesOnly() NucleiSDKOptions {
 		return nil
 	}
 }
+
+// UseSuppliedCatalog uses a supplied catalog
+func UseSuppliedCatalog(cat catalog.Catalog) NucleiSDKOptions {
```

Comment:
We can rename this as `WithCatalog` -tbh, I don't think this is necessary since the catalog is initialized internally.

Reply:
Is the use case described in https://github.com/projectdiscovery/nuclei/issues/5059 something that you want/use?

Reply:
Yes, I use this currently via my fork.

Rather than storing the templates on disk, the templates required for the scan can be shipped to the application, loaded into memory in a custom implementation of the `catalog.Catalog` interface, which can then discarded after the scan completes.

Reply:
I still think that we should rename it

Reply:
> I still think that we should rename it

I agree. I have updated the function.